ceto/ais_adapter.py

```diff
@@ -0,0 +1,557 @@
+        return 0.75
+    elif ceto_ship_type == "ferry-pax":
+        return 0.55
+    elif ceto_ship_type == "yacht":
+        return 0.6
+    elif ceto_ship_type.startswith("service"):
+        return 0.65
+    elif ceto_ship_type == "miscellaneous-fishing":
+        return 0.45
+
+    # else
+    raise ValueError("Invalid ceto shiptype %s was provided as input!", ceto_ship_type)
+
+
+def _guesstimate_design_draft(ais_draft: float, beam: float) -> float:
+    """Guesstimate the design draft
+
+    If the ais draft falls within the expected T/B ratio, use it directly,
+    otherwise estimate the design draft as the B/3.25
+
+    Args:
+        ais_draft (float): Draft from ais message 5 [m]
+        beam (float): Beam [m]
+
+    Returns:
+        float: Estimated design draft [m]
+    """
+    return ais_draft if 2.25 <= beam / ais_draft <= 3.75 else beam / 3.25
+
+
+def _guesstimate_design_speed(
+    length: float, ceto_ship_type: str, speed: float
+) -> float:
+    """Guesstimate the design speed
+
+    Assuming the there is a linear correlation between block coefficient
+    and design froude number. If the inputted actual speedis higher than
+    the estimated design spee it is used directly.
+
+    Args:
+        length (float): Length of vessel [m]
+        ceto_ship_type (str): The ceto ship type
+        speed (float): Speed of vessel [kn]
+
+    Returns:
+        float: Estimated design speed [kn]
+    """
+    g = 9.81
+    block_coefficient = _guesstimate_block_coefficient(ceto_ship_type)
+
+    blocks = [0.45, 0.8]
+    froude_numbers = [0.32, 0.145]
+
+    froude_number = np.interp(block_coefficient, blocks, froude_numbers)
+
+    design_speed = ms_to_knots(
+        (froude_number * math.sqrt(g * length)) / block_coefficient
+    )
+    return speed if speed > design_speed else design_speed
+
+
+def _guesstimate_number_of_engines(ceto_ship_type: str) -> int:
+    """Guesstimate number of engines
+
+    Self-explanatory and very simplistic approach...
+
+    Args:
+        ceto_ship_type (str): The ceto ship type
+
+    Returns:
+        int: The estimated number of engines
+    """
+    return 2 if ceto_ship_type in ("ferry-pax", "ferry-ropax") else 1
+
+
+def _guesstimate_engine_MCR(
+    ceto_ship_type: str, dwt: float, design_speed: float
+) -> float:
+    """Guesstimate the engine MCR based on deadweight and design speed
+
+    Sources:
+     - Cepowski, Tomasz. (2019). Regression Formulas for The Estimation of
+       Engine Total Power for Tankers, Container Ships and Bulk Carriers on
+       The Basis of Cargo Capacity and Design Speed. Polish Maritime Research.
+       26. 82-94. 10.2478/pomr-2019-0010.
+
+    Args:
+        ceto_ship_type (str): The ceto ship type
+        dwt (float): Deadweight tonnage [t]
+        design_speed (float): Design speed [kn]
+
+    Returns:
+        float: Estimated engine MCR [kW]
+    """
+    # Very crude model to differentiate between some (...) ship types
```

Comment:
Isnt that just the table taken from the Third IMO report.... (2014) ? But yes, that might be a better option.

Reply:
No, I will keep the existing implementation since it is usable for all sizes of ships. It may be more crude but will effectively allow us to perform the calculations for more ais targets.